AC_CHECK_FUNCS does not allow to inject includes (did ever alow that?) and the
only way to do that is by using AC_CHECK_DECLS. This generates a HAVE_DECL_fcn
define.

Upstream-Status: Backport
Signed-off-by: Andrei Gherzan <andrei@resin.io>

Index: openvpn-2.3.6/configure.ac
===================================================================
--- openvpn-2.3.6.orig/configure.ac
+++ openvpn-2.3.6/configure.ac
@@ -603,10 +603,10 @@ AC_SUBST([SOCKETS_LIBS])
 old_LIBS="${LIBS}"
 LIBS="${LIBS} ${SOCKETS_LIBS}"
 AC_CHECK_FUNCS([sendmsg recvmsg inet_ntop inet_pton])
-AC_CHECK_FUNCS(
+AC_CHECK_DECLS(
 	[res_init],
-	,
-	,
+	[],
+	[],
 	[[#include <resolv.h>]]
 )
 # Windows use stdcall for winsock so we cannot auto detect these
Index: openvpn-2.3.6/src/openvpn/socket.c
===================================================================
--- openvpn-2.3.6.orig/src/openvpn/socket.c
+++ openvpn-2.3.6/src/openvpn/socket.c
@@ -137,7 +137,7 @@ openvpn_getaddrinfo (unsigned int flags,
 
   ASSERT(res);
 
-#if defined(HAVE_RES_INIT)
+#if defined(HAVE_DECL_RES_INIT)
   res_init ();
 #endif
 
